### ISSUE TITLE
High variance of the result on DrugOOD.

### ISSUE THREAD

**Issue by panmianzhi** [yangnianzu0515/MoleOOD#3]
I run your code on DrugOOD for 5 times, but the result has very high variance. For example, the AUC-ROC on data_assay_ec50 are 71.41, 72.39, 61.93, 64.70, 67.24. 
I can't understand why this happen. Maybe I make mistakes about the hyper-parameters, and they are  as following:
<img width="444" alt="屏幕截图 2024-01-27 205012" src="https://github.com/yangnianzu0515/MoleOOD/assets/55686772/4f577b85-3f7e-4fea-9677-d6fa3e605513">
I saw the similar question in [the close issue](https://github.com/yangnianzu0515/MoleOOD/issues/2). Can you tell me the detailed value of the hyper-parameters on each dataset (except the random seed)?

**Comment by yangnianzu0515** [comment]
Thank you very much for your interest in our work.

We think the issue might be related to your selection of hyperparameters. We notice in a recent paper accepted by NeurIPS23, titled "Learning Invariant Molecular Representation in Latent Discrete Space", the variance on the EC50 dataset was not as significant as what you have experienced. 
![image](https://github.com/yangnianzu0515/MoleOOD/assets/51359992/525497ac-f071-445b-be9d-d0fb2fda79ff)
However, it's been about a year and a half since that work was published, and we cannot recall the specific parameters. We suspect that part of the reason for the discrepancy might be due to your setting of the 'environment number' hyperparameter to 20, which seems somewhat high.

We hope our response is of some assistance to you.

**Comment by panmianzhi** [comment]
Thank you very much! PS: actually I also run the code of _Learning Invariant Molecular Representation in Latent Discrete Space_ on DrugOOD, but the result is also not as good as their paper😂😂 Anyway, thank you for your reply!

**Comment by mxqmxqmxq** [comment]
@panmianzhi May I ask if you have read the [article](https://github.com/BUPT-GAMMA/Advancing-Molecule-Invariant-Representation-via-Privileged-Substructure-Identification) published in KDD 2024? It is of the same type as this one！
